File: list/src/main/java/ru/igormayachenkov/list/DItem.java
package ru.igormayachenkov.list;

////////////////////////////////////////////////////////////////////////////////////////////////
// DATA OBJECT: Item

public class DItem {
    public static final int     ITEM_STATE_CHECKED  = 0b0001;

    long        id;
    int         syncState;
    int         state;
    String      name;
    String      description;

    void changeState(){
        // Change
        if(state==ITEM_STATE_CHECKED)
            state = 0;
        else
            state = ITEM_STATE_CHECKED;
        // Save
        Data.instance().database.updateItemState(id,state);
    }
}

File: list/src/main/java/ru/igormayachenkov/list/data/DItem.kt
package ru.igormayachenkov.list.data

////////////////////////////////////////////////////////////////////////////////////////////////
// DATA OBJECT: Item
data class DItem(
        val id:Long
) {
    @JvmField
    var syncState = 0
    @JvmField
    var state = 0
    @JvmField
    var name: String? = null
    @JvmField
    var description: String? = null

    fun changeState() {
        // Change
        state = if (state == ITEM_STATE_CHECKED) 0 else ITEM_STATE_CHECKED
        // Save
        Database.updateItemState(id, state)
    }

    companion object {
        const val ITEM_STATE_CHECKED = 1
    }
}